cookie_consent/middleware.py

```diff
@@ -26,7 +31,13 @@ def process_response(self, request, response):
                 continue
             group_version = cookie_dic.get(cookie_group.varname, None)
             for cookie in cookie_group.cookie_set.all():
-                if cookie.name not in request.COOKIES:
+                if cookie.name not in str(request.COOKIES):
```

Comment:
checking based on the `str` representation of `request.COOKIES` doesn't seem right - `request.COOKIES` is a dict keyed by cookie name: https://docs.djangoproject.com/en/4.0/ref/request-response/#django.http.HttpRequest.COOKIES

Reply:
I had to do it this way because cookie.name is a string. You can't tell otherwise if the string object for cookie name is in a dictionary object. Unless there is a better way to do it that I don't know about?

Reply:
I need to dive more into the implementation details to figure out what's correct, but I'm certain that taking `str(request.COOKIES)` is not. If I have a cookie named `foo` in my config and some completely unrelated cookie (like the session ID) has that in the value, I have a problem now.

Example to illustrate this:

```py
>>> dummy = {"foo": "bar", "baz": "quux"}
>>> str(dummy)
"{'foo': 'bar', 'baz': 'quux'}"
>>> "bar" in str(dummy)
True
>>> "bar" in dummy
False
```
In this case, the `cookie.name == "bar"` would give a false positive.

Reply:
Right. That is why I came up with request.cookies.get() as an alternative in the above comment. This alternative works when testing on my system. Do you want me to change it? You should have edit access so that you can change it yourself though and I am okay with that - you seem to know best. So feel free to do so.